enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -117,9 +120,11 @@ impl<'a> Handler<'a> {
             self.ctx,
         );
 
-        self.block.msg.rep = Ok([ret.into(), 0usize.into()]).into();
+        let rep = Ok([ret.into(), 0usize.into()]);
 
-        ret
+        self.block.msg.rep = rep.into();
+
+        rep
```

Comment:
This is very close, but unfortunately incorrect.

The `ret` variable contains both the return contents and the errno. So we can't just return `Ok` for every syscall, because one might fail. You can see the logic we use in `sallyport` to handle this [here](https://github.com/enarx/enarx/blob/master/sallyport/src/lib.rs#L89-L117).

It may be simple enough just to convert the `Ok([ret, 0])` into a `Reply` and back again:

```suggestion
        self.block.msg.rep = rep.into();
        self.block.msg.rep.into()
```

... so long as the compiler doesn't try to optimize anything out incorrectly. I *think* we'll be okay.

Reply:
@npmccallum Happy to do it that way, though if there's some chance the compiler could introduce an error, maybe I should use `match` statements the way we do in `sallyport`. It's a bit more verbose but also probably more readable than converting twice. What do you think?

Reply:
Let's use the double conversion.